res/controllers/Numark-Scratch-scripts.js

```diff
@@ -3,36 +3,33 @@ var NumarkScratch = {};
 
 /*
  * USER CONFIGURABLE SETTINGS.
+ * Change settings in the preferences
  */
 
 // Defines the Beatloop Roll sizes for the 4 pads, for available values see:
 // https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatlooproll_X_activate
 // Default: [0.25, 0.5, 1, 2]
 NumarkScratch.autoLoopSizes = [
-    "0.25",
-    "0.5",
-    "1",
-    "2",
+    engine.getSetting("beatLoopRollsSize1") || 0.25,
+    engine.getSetting("beatLoopRollsSize2") || 0.5,
+    engine.getSetting("beatLoopRollsSize3") || 1,
+    engine.getSetting("beatLoopRollsSize4") || 2,
 ];
 
 // Defines how the Loop Encoder functions.
 // If 'true' the Encoder scrolls the library/loads track. Shift + Encoder manages looping.
 // If 'false' the Encoder manages looping. Shift + Encoder scrolls the library/loads track (Serato default).
 // Default: false
-NumarkScratch.invertLoopEncoderFunction = false;
+NumarkScratch.invertLoopEncoderFunction = !!engine.getSetting("invertLoopEncoderFunction");
 
-// Defines the bightness of button LEDs when they inactive.
-// '0x00' sets the LEDSs to completely off. '0x01 sets the LEDs to dim.
-// Default: 0x01 (dim)
-NumarkScratch.LOW_LIGHT = 0x01;
+// Define whether or not to keep LEDs dimmed if they are inactive.
+// 'true' will keep them dimmed, 'false' will turn them off. Default: true
+components.Button.prototype.off = !!engine.getSetting("inactiveLightsAlwaysBacklit");
```

Comment:
Is it safe to change the type of `components.Button.prototype.off` here? You might want to consider a ternary otherwise 

```suggestion
components.Button.prototype.off = !!engine.getSetting("inactiveLightsAlwaysBacklit") ? 0x1 : 0x0;
```

(Worth keeping `0x0` and `0x1` as constant, eg `NumarkScratch.NO_LIGHT` and `NumarkScratch.LOW_LIGHT` as well for readability)

Reply:
Done